pilot/pkg/model/push_context.go

```diff
@@ -723,11 +723,22 @@ func (ps *PushContext) DestinationRule(proxy *Proxy, service *Service) *Config {
 
 	// 4. if no public/private rule in calling proxy's namespace matched, and no public rule in the
 	// target service's namespace matched, search for any exported destination rule in the config root namespace
-	// NOTE: This does mean that we are effectively ignoring private dest rules in the config root namespace
 	if out := ps.getExportedDestinationRuleFromNamespace(ps.Mesh.RootNamespace, service.Hostname, proxy.ConfigNamespace); out != nil {
 		return out
 	}
 
+	// 5. Previously we skipped over private DR for the root namespace. If we still haven't found one, now we can check the private one.
+	// NOTE: This does mean that we are effectively giving destination rules in other namespace precedence over private dest rules in the config root namespace
+	if proxy.ConfigNamespace == ps.Mesh.RootNamespace {
```

Comment:
why cant we move this check to the top ? [sorry if this is naive as I dont remember this code anymore. ptsd]

Reply:
We have two indexes: one for things that are global, and one for things that are global or private. We do not want to select a global DR in root-namespace first, or we end up selecting a `*.cluster.local` rule with top precedence every time. 

So we can move it up to the top, but we need a new index for things that are *just* global. This is probably better behavior, I am leaning towards doing that

Reply:
Yeah, we can move it in parallel with **// 1. select destination rule from proxy config namespace** So that the proxy's config namespace takes priority over the service's

Reply:
done